### PR TITLE
API Updates for HA Custom Config Groups

### PR DESCRIPTION
**Pull request by riddhi89**

### CODE REVIEW

File: api-docs/api-operations/methods/patch-attach-configuration-group-to-ha-instance-version-accountid-ha-haid.rst [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+.. code::
+
+    PATCH /{version}/{accountId}/ha/{haId}
+
+Attaches a specified configuration group to the HA Instance (See :ref:`post-create-configuration-version-accountid-configurations`).
+
+.. note::
+   If the configuration group has non-dynamic configuration parameters, the HA instance will be put in a ``RESTART_REQUIRED`` state. To enable the parameters on all the nodes (source and replicas of the HA group), restart the HA instance (see :ref:`post-restart-ha-instance-version-accountid-ha-haid-action`.).
+
```
Comment (meker12) [inline api-docs/api-operations/methods/patch-attach-configuration-group-to-ha-instance-version-accountid-ha-haid.rst]: To enable the parameters on all the nodes (source and replicas of the HA group), :ref:`restart the HA instance <post-restart-ha-instance-version-accountid-ha-haid-action>`.

File: api-docs/api-operations/methods/patch-attach-configuration-group-to-ha-instance-version-accountid-ha-haid.rst [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+Attach Configuration Group to HA Instance 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code::
+
+    PATCH /{version}/{accountId}/ha/{haId}
+
+Attaches a specified configuration group to the HA Instance (See :ref:`post-create-configuration-version-accountid-configurations`).
+
```
Comment (meker12) [inline api-docs/api-operations/methods/patch-attach-configuration-group-to-ha-instance-version-accountid-ha-haid.rst]: Attaches a specified :ref:`configuration group <cdb-dg-concepts-configgroup>` to the HA Instance.

File: api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+Restart a HA Instance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code::
+
+    POST /{version}/{accountId}/ha/{haId}/action
+
+Restarts a HA Instance. When the HA instance is being restarted, it goes into a ``REBOOT`` state. It switches back to ``ACTIVE`` once the operation is complete.
+
```
Comment (meker12) [inline api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst]: Restarts an HA instance. During the restart process, the HA Instance goes into a ``REBOOT`` state. After the operation completes, the status changes  to ``ACTIVE``.

File: api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+.. warning::
+   When the HA Instance Group is restarted, auto-failover is turned off first (MHA stopped), mysql service is restarted on the replicas followed by the source and lastly auto-failover (MHA started) is turned on again. Refer to the Knowledge Center article for more details about MHA: `High Availability for Cloud Databases <https://www.rackspace.com/knowledge_center/article/high-availability-for-cloud-databases>`__.
+
```
Comment (meker12) [inline api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst]: When the HA Instance Group is started, auto-failover is turned off (MHA stopped), then the mysql replica nodes are restarted followed by the source nodes.  Finally, auto-failover is turned on again. 
turn off auto-failover (MHA stopped)


Revision note:  Not sure if "nodes" is correct here, but clarify what "replicas" and "source"  are.

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-b7d28d6e1c/docs/cloud-databases/v1/developer-guide/)

**Comment by riddhi89** [comment]
@meker12 ! thanks for the feedback..updated:)

File: api-docs/api-operations/methods/patch-remove-configuration-group-from-ha-instance-version-accountid-ha-haid.rst [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+.. code::
+
+    PATCH /{version}/{accountId}/ha/{haId}
+
+Detach configuration group from the HA Instance.
+
+.. note::
+   If the configuration group has non-dynamic configuration parameters, the HA instance will be put in a ``RESTART_REQUIRED`` state. To disable the parameters on all the nodes (source and replicas of the HA group), restart the HA instance (see :ref:`post-restart-ha-instance-version-accountid-ha-haid-action`.).
+
```
Comment (meker12) [inline api-docs/api-operations/methods/patch-remove-configuration-group-from-ha-instance-version-accountid-ha-haid.rst]: :ref:`restart the HA instance<post-restart-ha-instance-version-accountid-ha-haid-action>`.

Remove the "See" reference.

File: api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+Restart a HA Instance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code::
+
+    POST /{version}/{accountId}/ha/{haId}/action
+
+Restarts an HA instance. During the restart process, the HA Instance goes into a ``REBOOT state``. After the operation completes, the status changes to ``ACTIVE``.
+
```
Comment (meker12) [inline api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst]: a ``REBOOT`` state.

File: api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+.. warning::
+   When the HA Instance Group is started, auto-failover is turned off (MHA stopped), then the mysql service on the replica nodes is restarted followed by the source node. Finally, auto-failover is turned on again. Refer to the Knowledge Center article for more details about MHA: `High Availability for Cloud Databases <https://www.rackspace.com/knowledge_center/article/high-availability-for-cloud-databases>`__.
+
```
Comment (meker12) [inline api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst]: Replace "Refer to the ...." with the following:

See the :how-to:`High Availability for Cloud Databases <high-availability-for-cloud-databases>` article in the Rackspace How To collection.

File: api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+.. _post-restart-ha-instance-version-accountid-ha-haid-action.rst:
+
+Restart a HA Instance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (meker12) [inline api-docs/api-operations/methods/post-restart-ha-instance-version-accountid-ha-haid-action.rst]: Restart an HA Instance

**Comment by meker12** [comment]
@riddhi89  A couple of fixes.  Also, note that you can view the compiled content by clicking the Preview link in the PR comment.

**Comment by rackernexus** [comment]
:eyes: [Preview](https://staging.developer.rackspace.com/build-cd4aa152de/docs/cloud-databases/v1/developer-guide/)

**Comment by riddhi89** [comment]
updated! thanks:)

**Comment by rackernexus** [comment]
If you want to double-check your formatting, check [the preview](https://staging.developer.rackspace.com/build-110f32eb9a/docs/cloud-databases/v1/developer-guide/).

**Pull request merged**